language/move-lang/src/cfgir/locals/mod.rs

```diff
@@ -115,29 +115,26 @@ fn command(context: &mut Context, sp!(loc, cmd_): &Command) {
                     | LocalState::MaybeUnavailable { available, .. } => {
                         let ty = context.local_type(&local);
                         let kind = ty.value.kind(ty.loc);
-                        match &kind.value {
-                            Kind_::Unrestricted | Kind_::Affine => (),
-                            Kind_::Resource | Kind_::Unknown => {
-                                let verb = match (state, &kind.value) {
-                                    (LocalState::Unavailable(_), _) => unreachable!(),
-                                    (LocalState::Available(_), Kind_::Resource) => "contains",
-                                    _ => "might contain",
-                                };
-                                let available = *available;
-                                let stmt = match display_var(local.value()) {
-                                    DisplayVar::Tmp => {
-                                        "The resource is created but not used".to_owned()
-                                    }
-                                    DisplayVar::Orig(l) => format!(
-                                    "The local {} still {} a resource value due to this assignment",
+                        if kind.value.is_resourceful() {
+                            let verb = match (state, &kind.value) {
+                                (LocalState::Unavailable(_), _) => unreachable!(),
+                                (LocalState::Available(_), Kind_::Resource) => "contains",
+                                _ => "might contain",
+                            };
+                            let available = *available;
+                            let stmt = match display_var(local.value()) {
+                                DisplayVar::Tmp => {
+                                    "The resource is created but not used".to_owned()
+                                }
+                                DisplayVar::Orig(l) => format!(
+                                    "The local '{}' still {} a resource value due to this assignment",
```

Comment:
The message with "still might contain" sounds awkward to me. I suggest moving the "still" into the verb so that you can have "still contains" and "might still contain".